testing/bdd/test_required_metadata.py

```diff
@@ -0,0 +1,12 @@
+import xml.etree.ElementTree as ET
+from pytest_bdd import scenarios, then
+
+scenarios('features/metadata/ebuttd_required_metadata.feature')
+
+
+@then('the EBUTTD document contains a documentMetadata element <element_name> with value <element_value>')
+def then_ebuttd_metadata_contains(test_context, element_name, element_value):
+    document = test_context['ebuttd_document']
+    tree = ET.fromstring(document.get_xml())
+    elements = tree.findall('{http://www.w3.org/ns/ttml}head/{http://www.w3.org/ns/ttml}metadata/{urn:ebu:tt:metadata}documentMetadata/{urn:ebu:tt:metadata}%s' % element_name)
```

Comment:
Is this doing a proper XPath query? If so, you might replace `/{urn:ebu:tt:metadata}documentMetadata/` with `//` to get all descendants of the `tt/head/metadata` element that are `ebuttm:documentMetadata` elements, and be future-proof in testing that that the elements are present in any valid location.

Reply:
Good shout, I didn't know you could do that. It seems to work with both valid forms